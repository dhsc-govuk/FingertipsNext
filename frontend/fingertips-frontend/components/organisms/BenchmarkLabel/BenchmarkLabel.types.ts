import {
  BenchmarkComparisonMethod,
  BenchmarkOutcome,
} from '@/generated-sources/ft-api-client';

export type BenchmarkLabelGroupConfig = Partial<
  Record<BenchmarkComparisonMethod, BenchmarkLabelConfig>
>;

export type BenchmarkLabelConfig = Partial<
  Record<
    BenchmarkOutcome | 'default' | 'middleWithJudgement',
    BenchmarkLabelTypeConfig
  >
>;

export interface BenchmarkLabelTypeConfig {
  backgroundColor?: string;
  color?: string;
  border?: string;
  tint?: string;
}
