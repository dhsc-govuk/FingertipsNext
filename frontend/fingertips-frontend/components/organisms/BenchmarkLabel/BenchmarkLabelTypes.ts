export enum BenchmarkLabelType {
  HIGHER = 'Higher',
  LOWER = 'Lower',
  BETTER = 'Better',
  SIMILAR = 'Similar',
  LOWEST = 'Lowest',
  LOW = 'Low',
  MIDDLE = 'Middle',
  HIGH = 'High',
  HIGHEST = 'Highest',
  WORST = 'Worst',
  WORSE = 'Worse',
  BEST = 'Best',
  NOT_COMPARED = 'Not compared',
}

export type BenchmarkLabelGroupConfig = Partial<
  Record<BenchmarkLabelGroupType, BenchmarkLabelConfig>
>;

export type BenchmarkLabelConfig = Partial<
  Record<BenchmarkLabelType | 'default', BenchmarkLabelTypeConfig>
>;

export interface BenchmarkLabelTypeConfig {
  backgroundColor?: string;
  color?: string;
  border?: string;
  tint?: string;
}

export const enum BenchmarkLabelGroupType {
  RAG = 'rag',
  RAG_99 = 'rag_99',
  QUINTILES = 'quintiles',
  QUINTILES_WITH_JUDGEMENT = 'quintiles_with_judgement',
}

