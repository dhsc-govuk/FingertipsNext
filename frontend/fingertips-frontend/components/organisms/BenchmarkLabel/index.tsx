'use client';
import { Tag } from 'govuk-react';
import styled from 'styled-components';
import React from 'react';

export const enum BenchmarkLabelType {
  HIGHER = 'higher',
  LOWER = 'lower',
  BETTER = 'better',
  SIMILAR = 'similar',
  LOWEST = 'lowest',
  LOW = 'low',
  MIDDLE = 'middle',
  HIGH = 'high',
  HIGHEST = 'highest',
  WORST = 'worst',
  WORSE = 'worse',
  BEST = 'best',
  NOT_COMPARED = 'not_compared',
}

export const enum BenchmarkLabelGroupType {
  RAG = 'rag',
  QUINTILES = 'quintiles',
  QUINTILES_WITH_VALUE = 'quintiles_wv',
}

interface BenchmarkLabelProps {
  type?: BenchmarkLabelType | string;
  group?: BenchmarkLabelGroupType | string;
}

export const getDefaultBenchmarkTagStyle = (
  group: BenchmarkLabelGroupType,
  type: BenchmarkLabelType
) => {
  switch (group) {
    case BenchmarkLabelGroupType.RAG: {
      switch (type) {
        case BenchmarkLabelType.BETTER:
          return {
            backgroundColor: 'var(--other-green, #00703C)',
            tint: 'SOLID',
          };
        case BenchmarkLabelType.SIMILAR:
          return {
            backgroundColor: 'var(--other-yellow, #FD0)',
            color: 'var(--other-black, #0B0C0C)',
          };
        case BenchmarkLabelType.WORSE:
          return {
            backgroundColor: 'var(--other-red, #D4351C)',
            tint: 'SOLID',
          };
        case BenchmarkLabelType.LOWER:
          return {
            backgroundColor: 'var(--other-light-blue, #5694CA)',
            color: 'var(--other-black, #0B0C0C)',
          };
        case BenchmarkLabelType.HIGHER:
          return {
            backgroundColor: '#003078',
            tint: 'SOLID',
          };
        default:
          return {
            backgroundColor: 'transparent',
            color: 'var(--other-black, #0B0C0C)',
            border: '1px solid #0B0C0C',
          };
      }
    }
    case BenchmarkLabelGroupType.QUINTILES: {
      switch (type) {
        case BenchmarkLabelType.LOWEST:
          return {
            backgroundColor: '#E4DDFF',
            color: 'var(--other-black, #0B0C0C)',
          };

        case BenchmarkLabelType.LOW:
          return {
            backgroundColor: '#CBBEF4',
            color: 'var(--other-black, #0B0C0C)',
          };
        case BenchmarkLabelType.MIDDLE:
          return {
            backgroundColor: '#AA90EC',
            color: 'var(--other-black, #0B0C0C)',
          };
        case BenchmarkLabelType.HIGH:
          return {
            backgroundColor: '#8B60E2',
            tint: 'SOLID',
          };
        default:
          return {
            backgroundColor: '#6B33C3',
            tint: 'SOLID',
          };
      }
    }
    case BenchmarkLabelGroupType.QUINTILES_WITH_VALUE: {
      switch (type) {
        case BenchmarkLabelType.WORST:
          return {
            backgroundColor: '#D494C1',
            color: 'var(--other-black, #0B0C0C)',
          };
        case BenchmarkLabelType.WORSE:
          return {
            backgroundColor: '#BC6AAA',
            color: 'var(--other-black, #0B0C0C)',
          };
        case BenchmarkLabelType.MIDDLE:
          return {
            backgroundColor: '#A44596',
            tint: 'SOLID',
          };
        case BenchmarkLabelType.BETTER:
          return {
            backgroundColor: '#812972',
            tint: 'SOLID',
          };
        default:
          return {
            backgroundColor: '#561950',
            tint: 'SOLID',
          };
      }
    }
  }
};

export const BenchmarkTagStyle = styled(Tag)<{
  legendType: BenchmarkLabelType;
  group: BenchmarkLabelGroupType;
}>(({ legendType, group }) => {
  const theme = getDefaultBenchmarkTagStyle(group, legendType);
  return {
    padding: '5px 8px 4px 8px',
    alignItems: 'center',
    margin: '0.3125em',
    fontWeight: '300',
    textTransform: 'unset',
    ...theme,
  };
});

const getBenchmarkLabel = (
  type: BenchmarkLabelType,
  group: BenchmarkLabelGroupType
) => {
  switch (group) {
    case BenchmarkLabelGroupType.RAG:
      switch (type) {
        case BenchmarkLabelType.BETTER:
          return 'Better (95%)';
        case BenchmarkLabelType.SIMILAR:
          return 'Similar';
        case BenchmarkLabelType.WORSE:
          return 'Worse (95%)';
        case BenchmarkLabelType.LOWER:
          return 'Lower (95%)';
        case BenchmarkLabelType.HIGHER:
          return 'Higher (95%)';
        default:
          return 'Not compared';
      }

    case BenchmarkLabelGroupType.QUINTILES:
    case BenchmarkLabelGroupType.QUINTILES_WITH_VALUE: {
      const value = type.toLowerCase();
      return value[0].toUpperCase() + value.slice(1);
    }
    default:
      return '';
  }
};

export const BenchmarkLabel: React.FC<BenchmarkLabelProps> = ({
  type,
  group,
}) => {
  const legendType =
    (type as BenchmarkLabelType) ?? BenchmarkLabelType.NOT_COMPARED;
  const groupType =
    (group as BenchmarkLabelGroupType) ?? BenchmarkLabelGroupType.RAG;
  const label = getBenchmarkLabel(legendType, groupType);
  return (
    <BenchmarkTagStyle legendType={legendType} group={groupType}>
      {label}
    </BenchmarkTagStyle>
  );
};
