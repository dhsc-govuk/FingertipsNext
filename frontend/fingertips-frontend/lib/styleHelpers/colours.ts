export enum GovukColours {
  Red = '#D4351C',
  Yellow = '#FFDD00',
  Green = '#00703C',
  Blue = '#1D70B8',
  DarkBlue = '#003078',
  LightBlue = '#5694CA',
  Purple = '#4C2C92',
  Black = '#0B0C0C',
  DarkGrey = '#505A5F',
  MidGrey = '#B1B4B6',
  LightGrey = '#F3F2F1',
  White = '#FFFFFF',
  LightPurple = '#A285D1',
  BrightPurple = '#912B88',
  Pink = '#D53880',
  LightPink = '#F499BE',
  Orange = '#F46A25',
  Brown = '#B58840',
  LightGreen = '#85994B',
  Turquoise = '#28A197',
  DarkPink = '#801650',
  OtherLightPurple = '#6F72AF',
  CharcoalGray = '#3D3D3D',
  Female = '#5352BE',
  Male = '#57AEF8',
  DarkSlateGray = '#D7D7D7',
}

// govuk-react table border colour
// it is the old gds grey 2
// and it appears govuk-react hasn't been updated with the latest gds colours
export const BorderColour = '#BfC1C3';

export enum TagColours {
  GreyText = '#303436',
  GreyBackground = '#E1E2E3',
  // benchmarks
  DarkRed = '#831504',
  LightBlue = '#AFCFE5', //chart-lower99
  // trends
  YellowBackground = '#FFF5BB',
  YellowText = '#514311',
  GreenBackground = '#C4DED3',
  GreenText = '#00502D',
  RedBackground = '#F6C6BF',
  RedText = '#270D0B',
  LightBlueBackground = '#E3EFF6',
}

export enum QuintileColours {
  Lowest = '#E4DDFF',
  Low = '#CBBEF4',
  Middle = '#AA90EC',
  High = '#8B60E2',
  Highest = '#6B33C3',
  Worst = '#D494C1',
  Worse = '#BC6AAA',
  MiddleWithValue = '#A44596',
  Better = '#812972',
  Best = '#561950',
}

export function getTextColour(backgroundColour: string) {
  switch (backgroundColour) {
    case GovukColours.Green:
    case GovukColours.Red:
    case GovukColours.LightBlue:
    case GovukColours.DarkBlue:
    case GovukColours.Blue:
    case GovukColours.Black:

    case TagColours.DarkRed:

    case QuintileColours.High:
    case QuintileColours.Highest:
    case QuintileColours.MiddleWithValue:
    case QuintileColours.Better:
    case QuintileColours.Best:
      return GovukColours.White;

    default:
      return GovukColours.Black;
  }
}
