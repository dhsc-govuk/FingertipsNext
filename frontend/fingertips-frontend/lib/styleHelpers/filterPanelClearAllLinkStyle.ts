import styled from 'styled-components';
import { Link } from 'govuk-react';

export const StyledRightClearAllLink = styled(Link)<{ $enabled: boolean }>(
  ({ $enabled }) => ({
    display: 'flex',
    justifyContent: 'flex-end',
    fontSize: '19px',
    pointerEvents: $enabled ? 'none' : 'auto',
  })
);

export const StyledFilterWithClearAllLink = styled('div')({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  whiteSpace: 'nowrap',
});
