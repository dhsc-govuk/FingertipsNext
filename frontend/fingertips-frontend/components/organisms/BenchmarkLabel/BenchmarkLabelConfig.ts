import {
  GovukColours,
  QuintileColours,
  TagColours,
} from '@/lib/styleHelpers/colours';
import { BenchmarkLabelGroupConfig } from '@/components/organisms/BenchmarkLabel/BenchmarkLabelTypes';
import {
  BenchmarkComparisonMethod,
  BenchmarkOutcome,
  IndicatorPolarity,
} from '@/generated-sources/ft-api-client';

const similar = {
  backgroundColor: GovukColours.Yellow,
  color: GovukColours.Black,
};

const notCompared = {
  backgroundColor: 'transparent',
  color: GovukColours.Black,
  border: '1px solid #0B0C0C',
};

export const getBenchmarkTagStyle = (
  group: BenchmarkComparisonMethod,
  type: BenchmarkOutcome,
  polarity: IndicatorPolarity
) => {
  const groupConfig = benchmarkLabelGroupConfig[group];
  if (!groupConfig) return notCompared;

  // special case Middle is used by Quintiles with and without judgement but has different colours
  if (
    polarity !== IndicatorPolarity.NoJudgement &&
    type === BenchmarkOutcome.Middle
  ) {
    return groupConfig['middleWithJudgement'];
  }

  return groupConfig[type] ?? groupConfig.default;
};

export const benchmarkLabelGroupConfig: BenchmarkLabelGroupConfig = {
  [BenchmarkComparisonMethod.CIOverlappingReferenceValue95]: {
    default: notCompared,
    [BenchmarkOutcome.Similar]: similar,
    // RAG
    [BenchmarkOutcome.Better]: {
      backgroundColor: GovukColours.Green,
      tint: 'SOLID',
    },
    [BenchmarkOutcome.Worse]: {
      backgroundColor: GovukColours.Red,
      tint: 'SOLID',
    },
    // BOB
    [BenchmarkOutcome.Lower]: {
      backgroundColor: GovukColours.LightBlue,
    },
    [BenchmarkOutcome.Higher]: {
      backgroundColor: GovukColours.DarkBlue,
      tint: 'SOLID',
    },
  },

  [BenchmarkComparisonMethod.CIOverlappingReferenceValue99_8]: {
    default: notCompared,
    [BenchmarkOutcome.Similar]: similar,
    // RAG
    [BenchmarkOutcome.Better]: {
      backgroundColor: GovukColours.LightGreen,
      color: GovukColours.Black,
    },

    [BenchmarkOutcome.Worse]: {
      backgroundColor: TagColours.DarkRed,
      tint: 'SOLID',
    },
    // BOB
    [BenchmarkOutcome.Lower]: {
      backgroundColor: TagColours.LightBlue,
      color: GovukColours.Black,
    },
    [BenchmarkOutcome.Higher]: {
      backgroundColor: GovukColours.Blue,
      tint: 'SOLID',
    },
  },

  [BenchmarkComparisonMethod.Quintiles]: {
    default: { backgroundColor: QuintileColours.Highest, tint: 'SOLID' },
    // no Judgement
    [BenchmarkOutcome.Lowest]: {
      backgroundColor: QuintileColours.Lowest,
      color: GovukColours.Black,
    },
    [BenchmarkOutcome.Low]: {
      backgroundColor: QuintileColours.Low,
      color: GovukColours.Black,
    },
    [BenchmarkOutcome.Middle]: {
      backgroundColor: QuintileColours.Middle,
      color: GovukColours.Black,
    },
    [BenchmarkOutcome.High]: {
      backgroundColor: QuintileColours.High,
      tint: 'SOLID',
    },
    // judgement
    [BenchmarkOutcome.Worst]: {
      backgroundColor: QuintileColours.Worst,
      color: GovukColours.Black,
    },
    [BenchmarkOutcome.Worse]: {
      backgroundColor: QuintileColours.Worse,
      color: GovukColours.Black,
    },
    middleWithJudgement: {
      backgroundColor: QuintileColours.MiddleWithValue,
      tint: 'SOLID',
    },
    [BenchmarkOutcome.Better]: {
      backgroundColor: QuintileColours.Better,
      tint: 'SOLID',
    },
    [BenchmarkOutcome.Best]: {
      backgroundColor: QuintileColours.Best,
      tint: 'SOLID',
    },
  },
};
