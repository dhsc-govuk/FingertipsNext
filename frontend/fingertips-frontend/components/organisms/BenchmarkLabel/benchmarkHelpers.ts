import { BenchmarkLabelType } from '@/components/organisms/BenchmarkLabel/BenchmarkLabelTypes';
import {
  GetHealthDataForAnIndicatorComparisonMethodEnum,
  HealthDataPointBenchmarkComparisonIndicatorPolarityEnum,
} from '@/generated-sources/ft-api-client';

export const getBenchmarkType = (
  areaOfInterestLowerConfidence: number,
  areaOfInterestUpperConfidence: number,
  benchmarkValue: number,
  comparisonMethod: GetHealthDataForAnIndicatorComparisonMethodEnum,
  polarity?: HealthDataPointBenchmarkComparisonIndicatorPolarityEnum
): BenchmarkLabelType => {
  const comparisonValue = getComparisonValue(
    areaOfInterestLowerConfidence,
    areaOfInterestUpperConfidence,

    benchmarkValue
  );

  return getComparisonBenchmarkType(
    comparisonValue,
    comparisonMethod,
    polarity
  );
};

const getComparisonValue = (
  areaOfInterestLowerConfidence: number,
  areaOfInterestUpperConfidence: number,

  benchmarkValue: number
): -1 | 0 | 1 => {
  if (areaOfInterestUpperConfidence < benchmarkValue) return -1;
  if (areaOfInterestLowerConfidence > benchmarkValue) return 1;
  return 0;
};

const getComparisonBenchmarkType = (
  comparisonValue: -1 | 0 | 1,
  comparisonMethod: GetHealthDataForAnIndicatorComparisonMethodEnum,
  polarity?: HealthDataPointBenchmarkComparisonIndicatorPolarityEnum
) => {
  const comparisonValueAfterPolarity =
    comparisonValue *
    (polarity ===
    HealthDataPointBenchmarkComparisonIndicatorPolarityEnum.LowIsGood
      ? -1
      : 1);

  if (
    comparisonMethod === GetHealthDataForAnIndicatorComparisonMethodEnum.Rag
  ) {
    switch (comparisonValueAfterPolarity) {
      case -1:
        return BenchmarkLabelType.WORSE;
      case 1:
        return BenchmarkLabelType.BETTER;
      default:
        return BenchmarkLabelType.SIMILAR;
    }
  }

  switch (comparisonValueAfterPolarity) {
    case -1:
      return BenchmarkLabelType.LOWER;
    case 1:
      return BenchmarkLabelType.HIGHER;
    default:
      return BenchmarkLabelType.SIMILAR;
  }
};
