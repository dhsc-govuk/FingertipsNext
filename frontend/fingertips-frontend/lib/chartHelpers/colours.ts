import { GovukColours } from '../styleHelpers/colours';

export type ChartColours = GovukColours | UniqueChartColours;

export enum UniqueChartColours {
  ChartDarkBlue = '#12436D',
  OtherLightBlue = '#57AEF8',
}

// The ordering of this array determines the order in which the colours are
// used by Highcharts, so it should be preserved.
export const chartColours: ChartColours[] = [
  GovukColours.Yellow,
  GovukColours.Red,
  GovukColours.Blue,
  GovukColours.LightPink,
  GovukColours.Brown,
  UniqueChartColours.ChartDarkBlue,
  GovukColours.LightBlue,
  GovukColours.OtherLightPurple,
  GovukColours.Orange,
  GovukColours.LightPurple,
  GovukColours.DarkPink,
  GovukColours.Green,
  GovukColours.Pink,
  GovukColours.Purple,
];
