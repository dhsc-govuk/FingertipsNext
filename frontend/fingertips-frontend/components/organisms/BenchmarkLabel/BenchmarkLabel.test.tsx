import { render } from '@testing-library/react';
import { BenchmarkLabel } from '@/components/organisms/BenchmarkLabel/index';
import { QuintileColours } from '@/lib/styleHelpers/colours';
import {
  BenchmarkLabelGroupType,
  BenchmarkLabelType,
} from './BenchmarkLabelTypes';
import { getDefaultBenchmarkTagStyle } from '@/components/organisms/BenchmarkLabel/BenchmarkLabelConfig';

describe('testing the function getBenchmarkLegendColourStyle', () => {
  test.each([
    [
      BenchmarkLabelGroupType.RAG,
      BenchmarkLabelType.BETTER,
      {
        backgroundColor: 'var(--other-green, #00703C)',
        tint: 'SOLID',
      },
    ],

    [
      BenchmarkLabelGroupType.RAG,
      BenchmarkLabelType.SIMILAR,
      {
        backgroundColor: 'var(--other-yellow, #FFDD00)',
        color: 'var(--other-black, #0B0C0C)',
      },
    ],

    [
      BenchmarkLabelGroupType.RAG,
      BenchmarkLabelType.WORSE,
      {
        backgroundColor: 'var(--other-red, #D4351C)',
        tint: 'SOLID',
      },
    ],
    [
      BenchmarkLabelGroupType.RAG,
      BenchmarkLabelType.LOWER,
      {
        backgroundColor: 'var(--other-light-blue, #5694CA)',
      },
    ],

    [
      BenchmarkLabelGroupType.RAG,
      BenchmarkLabelType.HIGHER,
      {
        backgroundColor: '#003078',
        tint: 'SOLID',
      },
    ],
    [
      BenchmarkLabelGroupType.RAG,
      BenchmarkLabelType.NOT_COMPARED,
      {
        backgroundColor: 'transparent',
        color: '#0B0C0C',
        border: '1px solid #0B0C0C',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES,
      BenchmarkLabelType.LOWEST,
      {
        backgroundColor: QuintileColours.Lowest,
        color: 'var(--other-black, #0B0C0C)',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES,
      BenchmarkLabelType.LOW,
      {
        backgroundColor: '#CBBEF4',
        color: 'var(--other-black, #0B0C0C)',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES,
      BenchmarkLabelType.MIDDLE,
      {
        backgroundColor: '#AA90EC',
        color: 'var(--other-black, #0B0C0C)',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES,
      BenchmarkLabelType.HIGH,
      {
        backgroundColor: '#8B60E2',
        tint: 'SOLID',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES,
      BenchmarkLabelType.HIGHEST,
      {
        backgroundColor: '#6B33C3',
        tint: 'SOLID',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES_WITH_VALUE,
      BenchmarkLabelType.WORST,
      {
        backgroundColor: '#D494C1',
        color: 'var(--other-black, #0B0C0C)',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES_WITH_VALUE,
      BenchmarkLabelType.WORSE,
      {
        backgroundColor: '#BC6AAA',
        color: 'var(--other-black, #0B0C0C)',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES_WITH_VALUE,
      BenchmarkLabelType.MIDDLE,
      {
        backgroundColor: '#A44596',
        tint: 'SOLID',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES_WITH_VALUE,
      BenchmarkLabelType.BETTER,
      {
        backgroundColor: '#812972',
        tint: 'SOLID',
      },
    ],
    [
      BenchmarkLabelGroupType.QUINTILES_WITH_VALUE,
      BenchmarkLabelType.BEST,
      {
        backgroundColor: '#561950',
        tint: 'SOLID',
      },
    ],
  ])(
    'returns correct style for %s group and %s type',
    (group, type, expected) => {
      const result = getDefaultBenchmarkTagStyle(
        group as BenchmarkLabelGroupType,
        type as BenchmarkLabelType
      );
      expect(result).toEqual(expected);
    }
  );
});
/* The component test for the UI */

describe('Testing the BenchmarkLabel Component', () => {
  test('renders with default props', () => {
    const { getByText, container } = render(<BenchmarkLabel />);
    expect(getByText('Not compared')).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('background-color:transparent');
    expect(container.firstChild).toHaveStyle('color:#0B0C0C');
  });

  test('renders with a specified type and group', () => {
    const { getByText } = render(
      <BenchmarkLabel
        type={BenchmarkLabelType.BETTER}
        group={BenchmarkLabelGroupType.RAG}
      />
    );
    expect(getByText('Better')).toBeInTheDocument();
  });

  test('applies correct styles for RAG group and HIGHER type', () => {
    const { container } = render(
      <BenchmarkLabel
        type={BenchmarkLabelType.HIGHER}
        group={BenchmarkLabelGroupType.RAG}
      />
    );

    expect(container.firstChild).toHaveStyle('background-color: #003078');
  });

  test('applies correct styles for QUINTILE group and LOW type', () => {
    const { container } = render(
      <BenchmarkLabel
        type={BenchmarkLabelType.LOW}
        group={BenchmarkLabelGroupType.QUINTILES}
      />
    );
    expect(container.firstChild).toHaveStyle('background-color: #CBBEF4');
  });

  test('applies correct styles for OTHERS group and WORST type', () => {
    const { container } = render(
      <BenchmarkLabel
        type={BenchmarkLabelType.WORST}
        group={BenchmarkLabelGroupType.QUINTILES_WITH_VALUE}
      />
    );
    expect(container.firstChild).toHaveStyle('background-color: #D494C1');
  });

  it('snapshot testing of the UI', () => {
    const container = render(
      <BenchmarkLabel
        type={BenchmarkLabelType.WORST}
        group={BenchmarkLabelGroupType.QUINTILES_WITH_VALUE}
      />
    );
    expect(container.asFragment()).toMatchSnapshot();
  });
});
