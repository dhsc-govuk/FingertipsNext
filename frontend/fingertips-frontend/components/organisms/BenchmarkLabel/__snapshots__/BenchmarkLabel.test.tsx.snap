// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the BenchmarkLabel Component  snapshot testing of the UI 1`] = `
<DocumentFragment>
  .c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 14px;
  line-height: 1;
  display: inline-block;
  padding-top: 5px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background-color: #1d70b8;
  color: #ffffff;
}

.c1 {
  display: inline-block;
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 300;
  font-style: normal;
  line-height: 16px;
  margin: 0.0225em;
  background-color: #D494C1;
  color: var(--other-black,#0B0C0C);
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 16px;
    line-height: 1;
  }
}

<strong
    class="c0 c1"
  >
    Worst
  </strong>
</DocumentFragment>
`;
